SG14/ring.h

```diff
@@ -108,7 +108,15 @@ namespace sg14
 		template <bool C>
 		bool operator==(const ring_iterator<Ring, C>& rhs) const noexcept;
 		template <bool C>
+		bool operator!=(const ring_iterator<Ring, C>& rhs) const noexcept;
+		template <bool C>
 		bool operator<(const ring_iterator<Ring, C>& rhs) const noexcept;
```

Comment:
@mattreecebentley commented:
> Does < or > even make sense in a ring?

I believe it does. `x < y` should be true iff `x - y < 0` is true, and I suppose you'd agree that `x - y` ought to be well-defined in a ring, right?

An alternative approach would be to give up on RandomAccessIterator and just have ring_iterator be a BidirectionalIterator, in which case we could omit the < <= > >= comparison operators; but I don't see any reason we *should* do that. RandomAccessIterator seems like the natural category to me.